src/core/wee-command.c

```diff
@@ -5344,7 +5344,14 @@ COMMAND_CALLBACK(secure)
         }
         else
         {
+            if (temp_secure_passphrase == NULL || strcmp (temp_secure_passphrase, argv_eol[2]))
+            {
+                temp_secure_passphrase = strdup (argv_eol[2]);
+                gui_chat_printf (NULL, "Run the passphrase command again to confirm!");
+                return WEECHAT_RC_OK;
+            }
             secure_passphrase = strdup (argv_eol[2]);
+            free(temp_secure_passphrase);
```

Comment:
This `free` is only done when the user finally enters the passphrase correctly the second time. The control flow does not reach this `free` to handle the leak in described above.

Also, WeeChat style would be to have a space between `free` and `(`.